### DISCUSSION TITLE
请教，本优选IP工具和works专用IP优选工具有什么区别？

### DISCUSSION THREAD

**Discussion by denniszzg** [XIU2/CloudflareSpeedTest#447] [提问] [answered]
背景：

网上下的works专用ip优选工具，命令 `CloudflareST.exe -tp 443 -url https://jp.cloudflarest.link`，选出的IP延迟低，速度快。查了一下IP运营商不是cloudflare，是 Oracle Corporation, Oracle Cloud Infrastructure (ap-seoul-1)

1. 在v2rayN服务器中将该IP填入地址，如果是VLESS协议（vless+ws+tls+nginx前置），可以翻墙，但日志中出现

```
[Warning] app/proxyman/outbound: failed to process outbound traffic > proxy/vless/outbound: connection ends > proxy/vless/outbound: failed to write A request payload > transport/internet/websocket: failed to dial WebSocket > transport/internet/websocket: failed to dial to (wss://该优选IP地址/fq/):  > tls: failed to verify certificate: x509: certificate is valid for www.ohmybobo.online, not xxx (此处是我服务器的域名)
```
2. 如果是替换成VMESS协议，其他不变，日志中不出现上述错误

搞不懂什么意思，求解惑。

1. works专用IP优选工具优选出的IP普遍比我用本仓库的优选工具选出的IP速度快，但查询IP的运营商都是aliyun、oracle、aws之类。我的疑惑：不是CF的CDN IP，为啥也能当优选IP使用在我的节点上呢？

2. 搜出来的IP是别人搭建的cloudflare workers节点吗？比如本例，www.ohmybobo.online

3. 我填写这个IP，是否意味着我的翻墙流量都经过了别人的服务器呢？这岂不是大大不妥？

**Comment by denniszzg** [comment]
我按照XIU2大佬的方法，自建了Cloudflare Workers 文件反代，搜出来的IP都是CF的IP
而 works专用IP优选工具选出来的却不是CF IP (-url https://jp.cloudflarest.link)
优选ip这背后到底是什么原理呢？不是同一个公司的cdn之间才应该有效吗？

**Comment by XIU2** [comment, marked as answer]
`-url` 是指定下载测速地址的，要测试的 IP 段范围默认是 `ip.txt` 文件，根据你的描述，估计是别人整理了一些 IP 段数据放到 `ip.txt` 文件内，然后和我的 CloudflareST 打包到一起提供给你们使用的。

我的 CloudflareST 内的 `ip.txt` 写的是 Cloudflare 官方提供的 IP 段数据，而你说的这些 IP 归属者都是 aliyun、oracle、aws 等等的，那肯定不是 Cloudflare 自己的 IP，而是别人搭建的 **反向代理/端口转发（后者也叫中转）**，不过大概率是扫到了别人没有做访问限制的服务器，你可以看看这个 #192

**Comment by denniszzg** [comment]
明白了，原来还是那些个人搭建的CF反代IP，我还以为是新方法呢。我看了ip.txt，确实如您所说，不是cf的ip段，是他自己整理的ip地址。这个ip_scan项目之前用过一下，觉得不稳定就没用了。
感谢解惑。
